### ISSUE TITLE
BUG: Search is case sensitive

### ISSUE THREAD

**Issue by hero101** [cherrytwist/Server#4272]
**Describe the bug**
"Capitals are defining search results: when I search for 'denise', I don't get any search results for contributors. When I search with capital for 'Denise', I do get the results. "

**To Reproduce**
1. Go to Home page
2. Click on Search
3. Search for 'denise'
4. No results for Contributor Denise Larsson
5. Search for 'Denise'
6. Results are found Contributor for Denise Larsson

**Expected behavior**
Search is not case sensitive

**Comment by hero101** [comment]
@SimoneZaza @techsmyth @larssondenise @DonnaDuiker 
After another pass over the problem, I have some findings.

### TL;DR
Case sensitivity gives more meaning and semantics to terms, so you would match terms in a specific context rather than the word itself context-free. Context-free will produce a lot of results, which might reduce the overall experience.
- We can play around with how we search, for example having the terms duplicated in both cases 🤷‍♂️ or using another ranking strategy.
- **Even better would be to have separate queries per result type. That can be implemented in a few ways but the overall experience would be much much better.**

Full analysis below...

---
From the user's POV, it seems like the thing they are searching for is not found, so I had a look what are the raw results returned from Elastic.
Certain search terms with different casing can generate different results or coincidentally the same results but with different scores. This is caused by the term matching with different fields and texts and is scored differently based on the context - for example, a term matched in text is case-insensitive but is scored more harshly than a term matched in a keyword like first, last name or nameID. To not overload the system we have a cut out point of 25 - so we get the top 25 results, and unfortunately, the most valuable thing you can use were below the threshold...

Concrete example:
> Searching for "Alkemio" in organizations will match the organization "Alkemio BV" in:
> - profile.displayName.text (4.206871)
> ... and will be scored with `4.206871` since it was matched in text only.

> Searching for "alkemio" in organizations will match the organization "Alkemio BV" in:
> - nameID (4.7125287)
> - profile.displayName.text (4.206871)
> ... and will be scored at `8.919399` since it was matched more than once.

The problem IMO is not only low vs upper case - casing can also influence the meaning and the semantics.

**Comment by SimoneZaza** [comment]
As discussed, we will continue by searching for 25 results per category instead of 25 in total.

**Comment by SimoneZaza** [comment]
As discussed, this is quite big, so there is a new epic to approach all that's needed:  #1496
